source/common/runtime/runtime_impl.h

```diff
@@ -143,10 +143,12 @@ class DiskLayer : public OverrideLayerImpl, Logger::Loggable<Logger::Id::runtime
     DIR* dir_;
   };
 
-  void walkDirectory(const std::string& path, const std::string& prefix);
+  void walkDirectory(const std::string& path, const std::string& prefix, uint32_t depth);
 
   const std::string path_;
   Api::OsSysCalls& os_sys_calls_;
+  // Maximum recursion depth for walkDirectory().
+  const uint32_t MaxWalkDepth = 4;
```

Comment:
I would go higher here. This might even be about where we are with our normal topology. Maybe 20?